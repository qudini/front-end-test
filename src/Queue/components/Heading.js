// components/Heading.js
import React from 'react';
import styled from 'styled-components';

const StyledHeading = styled.div`
  font-weight: bold;
  margin-bottom: 10px;
  color: darkslategrey;
  h1 {
    font-size: 2em;
  }

  h2 {
    font-size: 1.5em;
  }

  h3 {
    font-size: 1.2em;
  }
`;

const Heading = ({ level = 1, children }) => {
    const HeadingElement = `h${level}`;
    return <StyledHeading as={HeadingElement}>{children}</StyledHeading>;
};

export default Heading;
