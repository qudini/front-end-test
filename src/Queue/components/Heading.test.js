import React from 'react';
import { render, screen } from '@testing-library/react';
import Heading from './Heading';
import {axe} from "jest-axe";

describe('Heading Component', () => {
    it('renders with default styles', () => {
        render(<Heading>Test Heading</Heading>);

        const headingElement = screen.getByText('Test Heading');

        // Check if the heading is rendered
        expect(headingElement).toBeInTheDocument();

        // Check styles using jest-styled-components
        expect(headingElement).toHaveStyleRule('font-weight', 'bold');
        expect(headingElement).toHaveStyleRule('margin-bottom', '10px');
        expect(headingElement).toHaveStyleRule('color', 'darkslategrey');
    });

    it('renders different heading levels', () => {
        render(
            <>
                <Heading level={1}>Heading 1</Heading>
                <Heading level={2}>Heading 2</Heading>
                <Heading level={3}>Heading 3</Heading>
            </>
        );

        // Check if each heading level is rendered
        expect(screen.getByText('Heading 1').tagName).toBe('H1');
        expect(screen.getByText('Heading 2').tagName).toBe('H2');
        expect(screen.getByText('Heading 3').tagName).toBe('H3');
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Heading>Snapshot Test</Heading>);

        // Check if the component matches the snapshot
        expect(asFragment()).toMatchSnapshot();
    });

    it('is accessible', async () => {
        const {container} = render(<Heading>Heading</Heading>);

        const results = await axe(container);

        // Expect no violations
        expect(results).toHaveNoViolations();
    });
});
