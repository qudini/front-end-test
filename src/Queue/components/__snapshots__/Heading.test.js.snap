// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: bold;
  margin-bottom: 10px;
  color: darkslategrey;
}

.c0 h1 {
  font-size: 2em;
}

.c0 h2 {
  font-size: 1.5em;
}

.c0 h3 {
  font-size: 1.2em;
}

<h1
    class="c0"
  >
    Snapshot Test
  </h1>
</DocumentFragment>
`;
